### ISSUE TITLE
providers stop bidding

### ISSUE THREAD

**Issue by andy108369** [ovrclk/akash#1613]
Providers stop seeing the deployment orders.

There is just 1 provider which bids now,

- 1 CPU / 1Gi memory / 1Gi Storage deployment:

```
profiles:
  compute:
    ssh:
      resources:
        cpu:
          units: 1
        memory:
          size: 1Gi
        storage:
          size: 1Gi
  placement:
    akash:
      pricing:
        ssh:
          denom: uakt
          amount: 10000
```

- Checking the bids

```
akash-tools/cli-booster[https://rpc.akash.forbole.com:443][default][6284565-1-1]$ akash_accept 
AKASH_PROVIDER was not set so let's try to pick from what's available for your deployment.
	rate	monthly	usd	dseq/oseq/gseq	provider					host
0>	23.21	9.43	$3.23	6284565/1/1	akash1vppf2922vuzgxc2jgetxtx7uxqfss59e7gha3g	provider-la-1.maxihost.com:8443	
Choose your bid from the list [0]: ^C
```

- `akash events`

> Running on my provider (`provider.akash.pro`) which uses my RPC running in K8s.

Can see there is just 1 provider which bids now `akash1vppf2922vuzgxc2jgetxtx7uxqfss59e7gha3g`:

```
root@node1:~# date
Sun Jun 12 09:19:06 UTC 2022

root@node1:~# export AKASH_NODE=http://akash-node-1.akash-services.svc.cluster.local:26657
root@node1:~# akash events

{
  "context": {
    "module": "deployment",
    "action": "deployment-created"
  },
  "id": {
    "owner": "akash1h24fljt7p0nh82cq0za0uhsct3sfwsfu9w3c9h",
    "dseq": 6284565
  },
  "version": "yVFDGsFmS+09WClkNc5c7jjdORc9MR6LQcBL79hH6f8="
}
{
  "context": {
    "module": "market",
    "action": "order-created"
  },
  "id": {
    "owner": "akash1h24fljt7p0nh82cq0za0uhsct3sfwsfu9w3c9h",
    "dseq": 6284565,
    "gseq": 1,
    "oseq": 1
  }
}

{
  "context": {
    "module": "market",
    "action": "bid-created"
  },
  "id": {
    "owner": "akash1h24fljt7p0nh82cq0za0uhsct3sfwsfu9w3c9h",
    "dseq": 6284565,
    "gseq": 1,
    "oseq": 1,
    "provider": "akash1vppf2922vuzgxc2jgetxtx7uxqfss59e7gha3g"
  },
  "price": {
    "denom": "uakt",
    "amount": "23.214080000000000000"
  }
}
```

- RPC is synced

```
root@node1:~# akash status | jq
{
  "NodeInfo": {
    "protocol_version": {
      "p2p": "8",
      "block": "11",
      "app": "0"
    },
    "id": "2b19780ea3de3553f03847a397ac2dfda1d2e89e",
    "listen_addr": "tcp://0.0.0.0:26656",
    "network": "akashnet-2",
    "version": "0.34.19",
    "channels": "40202122233038606100",
    "moniker": "mynode-1",
    "other": {
      "tx_index": "on",
      "rpc_address": "tcp://0.0.0.0:26657"
    }
  },
  "SyncInfo": {
    "latest_block_hash": "52C1E559B404D41008BA933B27BD19F1FD89C9CCC0212FF1FFE4FD28A1E22CAD",
    "latest_app_hash": "5067E07A68A4C4DB90398D5EAB34A35EDA378EBDEEE2AD24D6242BCA48EAF299",
    "latest_block_height": "6284609",
    "latest_block_time": "2022-06-12T09:23:32.434963189Z",
    "earliest_block_hash": "B587517415CE80AB9555B384F44A5453116291E799785C50E58B731D28BDB5B3",
    "earliest_app_hash": "BA05AA2268A53A3E77C41CD3F4D50889F3877026FC592CC2E850BC7011A1E78E",
    "earliest_block_height": "6258001",
    "earliest_block_time": "2022-06-10T12:19:46.015092555Z",
    "catching_up": false
  },
  "ValidatorInfo": {
    "Address": "61BEA7F2AA6DC97EF09FAC6CBCFFBD18D35FC2C0",
    "PubKey": {
      "type": "tendermint/PubKeyEd25519",
      "value": "XPf+uG09snTT2UJh8ZPX95zitwIaccq05I58QDl4slk="
    },
    "VotingPower": "0"
  }
}
```

**Comment by andy108369** [comment]
- RPC has 0 unconfirmed TXs:
```
root@node1:~# curl http://akash-node-1.akash-services.svc.cluster.local:26657/unconfirmed_txs
{
  "jsonrpc": "2.0",
  "id": -1,
  "result": {
    "n_txs": "0",
    "total": "0",
    "total_bytes": "0",
    "txs": []
  }
}
```

- provider logs aren't showing anything related to the deployment order:

```
root@node1:~# kubectl -n akash-services logs $(kubectl -n akash-services get pods -l app=akash-provider --output jsonpath='{.items[0].metadata.name}') --tail=10 -f
D[2022-06-12|09:27:25.924] running check                                module=provider-cluster cmp=service cmp=deployment-manager lease=akash1djnh28gk83k9kewfcgn55ah3d7zdrpfy4uc7dg/6219782/1/1/akash1nxq8gmsw2vlz3m68qvyvcf3kh6q269ajvqw6y0 manifest-group=akash cmp=deployment-monitor attempt=1
I[2022-06-12|09:27:25.940] check result                                 module=provider-cluster cmp=service cmp=deployment-manager lease=akash1djnh28gk83k9kewfcgn55ah3d7zdrpfy4uc7dg/6219782/1/1/akash1nxq8gmsw2vlz3m68qvyvcf3kh6q269ajvqw6y0 manifest-group=akash cmp=deployment-monitor ok=true attempt=1
D[2022-06-12|09:27:28.879] running check                                module=provider-cluster cmp=service cmp=deployment-manager lease=akash130xhrhruhmv65w3awpgzylqkhzj7km5jgfyez9/6254087/1/1/akash1nxq8gmsw2vlz3m68qvyvcf3kh6q269ajvqw6y0 manifest-group=akash cmp=deployment-monitor attempt=1
I[2022-06-12|09:27:28.889] check result                                 module=provider-cluster cmp=service cmp=deployment-manager lease=akash130xhrhruhmv65w3awpgzylqkhzj7km5jgfyez9/6254087/1/1/akash1nxq8gmsw2vlz3m68qvyvcf3kh6q269ajvqw6y0 manifest-group=akash cmp=deployment-monitor ok=true attempt=1
D[2022-06-12|09:27:31.026] running check                                module=provider-cluster cmp=service cmp=deployment-manager lease=akash1angnwmnsv5vkp4jpgkdcunar8efpvennxwwcev/6194991/1/1/akash1nxq8gmsw2vlz3m68qvyvcf3kh6q269ajvqw6y0 manifest-group=tezos cmp=deployment-monitor attempt=1
I[2022-06-12|09:27:31.034] check result                                 module=provider-cluster cmp=service cmp=deployment-manager lease=akash1angnwmnsv5vkp4jpgkdcunar8efpvennxwwcev/6194991/1/1/akash1nxq8gmsw2vlz3m68qvyvcf3kh6q269ajvqw6y0 manifest-group=tezos cmp=deployment-monitor ok=true attempt=1
D[2022-06-12|09:27:33.016] running check                                module=provider-cluster cmp=service cmp=deployment-manager lease=akash1djnh28gk83k9kewfcgn55ah3d7zdrpfy4uc7dg/6277866/1/1/akash1nxq8gmsw2vlz3m68qvyvcf3kh6q269ajvqw6y0 manifest-group=akash cmp=deployment-monitor attempt=1
I[2022-06-12|09:27:33.023] check result                                 module=provider-cluster cmp=service cmp=deployment-manager lease=akash1djnh28gk83k9kewfcgn55ah3d7zdrpfy4uc7dg/6277866/1/1/akash1nxq8gmsw2vlz3m68qvyvcf3kh6q269ajvqw6y0 manifest-group=akash cmp=deployment-monitor ok=true attempt=1
I[2022-06-12|09:27:35.243] syncing sequence                             cmp=client/broadcaster local=20472 remote=20472
D[2022-06-12|09:27:37.110] running check                                module=provider-cluster cmp=service cmp=deployment-manager lease=akash1djnh28gk83k9kewfcgn55ah3d7zdrpfy4uc7dg/6219782/1/1/akash1nxq8gmsw2vlz3m68qvyvcf3kh6q269ajvqw6y0 manifest-group=akash cmp=deployment-monitor attempt=1
I[2022-06-12|09:27:37.119] check result                                 module=provider-cluster cmp=service cmp=deployment-manager lease=akash1djnh28gk83k9kewfcgn55ah3d7zdrpfy4uc7dg/6219782/1/1/akash1nxq8gmsw2vlz3m68qvyvcf3kh6q269ajvqw6y0 manifest-group=akash cmp=deployment-monitor ok=true attempt=1
D[2022-06-12|09:27:37.279] running check                                module=provider-cluster cmp=service cmp=deployment-manager lease=akash1gqh69dd54grj48h9650yfjvt73chmxr6x9dxx7/5804289/1/1/akash1nxq8gmsw2vlz3m68qvyvcf3kh6q269ajvqw6y0 manifest-group= cmp=deployment-monitor attempt=1
I[2022-06-12|09:27:37.289] check result                                 module=provider-cluster cmp=service cmp=deployment-manager lease=akash1gqh69dd54grj48h9650yfjvt73chmxr6x9dxx7/5804289/1/1/akash1nxq8gmsw2vlz3m68qvyvcf3kh6q269ajvqw6y0 manifest-group= cmp=deployment-monitor ok=true attempt=1
D[2022-06-12|09:27:37.313] cluster resources                            module=provider-cluster cmp=service cmp=inventory-service dump="{\"nodes\":[{\"name\":\"node1\",\"allocatable\":{\"cpu\":11800,\"memory\":66714087424,\"storage_ephemeral\":459931859412},\"available\":{\"cpu\":2550,\"memory\":50966499328,\"storage_ephemeral\":239680567764}}],\"total_allocatable\":{\"cpu\":11800,\"memory\":66714087424,\"storage_ephemeral\":459931859412,\"storage\":{\"beta3\":972856819712}},\"total_available\":{\"cpu\":2550,\"memory\":50966499328,\"storage_ephemeral\":239680567764,\"storage\":{\"beta3\":972856808630}}}\n"
D[2022-06-12|09:27:40.546] running check                                module=provider-cluster cmp=service cmp=deployment-manager lease=akash130xhrhruhmv65w3awpgzylqkhzj7km5jgfyez9/6254087/1/1/akash1nxq8gmsw2vlz3m68qvyvcf3kh6q269ajvqw6y0 manifest-group=akash cmp=deployment-monitor attempt=1
I[2022-06-12|09:27:40.564] check result                                 module=provider-cluster cmp=service cmp=deployment-manager lease=akash130xhrhruhmv65w3awpgzylqkhzj7km5jgfyez9/6254087/1/1/akash1nxq8gmsw2vlz3m68qvyvcf3kh6q269ajvqw6y0 manifest-group=akash cmp=deployment-monitor ok=true attempt=1
D[2022-06-12|09:27:44.761] running check                                module=provider-cluster cmp=service cmp=deployment-manager lease=akash1angnwmnsv5vkp4jpgkdcunar8efpvennxwwcev/6194991/1/1/akash1nxq8gmsw2vlz3m68qvyvcf3kh6q269ajvqw6y0 manifest-group=tezos cmp=deployment-monitor attempt=1
I[2022-06-12|09:27:44.769] check result                                 module=provider-cluster cmp=service cmp=deployment-manager lease=akash1angnwmnsv5vkp4jpgkdcunar8efpvennxwwcev/6194991/1/1/akash1nxq8gmsw2vlz3m68qvyvcf3kh6q269ajvqw6y0 manifest-group=tezos cmp=deployment-monitor ok=true attempt=1
I[2022-06-12|09:27:45.243] syncing sequence                             cmp=client/broadcaster local=20472 remote=20472
D[2022-06-12|09:27:46.120] running check                                module=provider-cluster cmp=service cmp=deployment-manager lease=akash1djnh28gk83k9kewfcgn55ah3d7zdrpfy4uc7dg/6277866/1/1/akash1nxq8gmsw2vlz3m68qvyvcf3kh6q269ajvqw6y0 manifest-group=akash cmp=deployment-monitor attempt=1
I[2022-06-12|09:27:46.130] check result                                 module=provider-cluster cmp=service cmp=deployment-manager lease=akash1djnh28gk83k9kewfcgn55ah3d7zdrpfy4uc7dg/6277866/1/1/akash1nxq8gmsw2vlz3m68qvyvcf3kh6q269ajvqw6y0 manifest-group=akash cmp=deployment-monitor ok=true attempt=1
```

- syncing sequence stalled at `20472`:

```
root@node1:~# kubectl -n akash-services logs $(kubectl -n akash-services get pods -l app=akash-provider --output jsonpath='{.items[0].metadata.name}') | grep 'syncing sequence' | head -2
I[2022-06-12|00:34:25.243] syncing sequence                             cmp=client/broadcaster local=20472 remote=20472
I[2022-06-12|00:34:35.243] syncing sequence                             cmp=client/broadcaster local=20472 remote=20472

root@node1:~# kubectl -n akash-services logs $(kubectl -n akash-services get pods -l app=akash-provider --output jsonpath='{.items[0].metadata.name}') | grep 'syncing sequence' | tail -2
I[2022-06-12|09:28:15.243] syncing sequence                             cmp=client/broadcaster local=20472 remote=20472
I[2022-06-12|09:28:25.243] syncing sequence                             cmp=client/broadcaster local=20472 remote=20472
```

- running a TX using the provider's key:

```
root@node1:~# kubectl -n akash-services exec -ti $(kubectl -n akash-services get pods -l app=akash-provider --output jsonpath='{.items[0].metadata.name}') -- bash

root@akash-provider-545b8cf57b-pw6bp:/# akash tx deployment create deploy.yaml
{"height":"6284671","txhash":"FD805919EA5298A75CDEF63237A85EBF4C3CCD6111C70DDFDC97DEE05982D23F","codespace":"","code":0,"data":"0A2F0A2D2F616B6173682E6465706C6F796D656E742E763162657461322E4D73674372656174654465706C6F796D656E74",..........,{\"key\":\"dseq\",\"value\":\"6284669\"},.......

root@akash-provider-545b8cf57b-pw6bp:/# akash tx deployment close --dseq 6284669 
{"height":"6284678","txhash":"F365ABC1320E0A12729281DBB784649E02E32FCFE1ACB94919C42B7DCD3BA2EE","codespace":"","code":0,.......
````

- account sequence is still at `20474` and hasn't moved forward:

```
root@akash-provider-545b8cf57b-pw6bp:/# akash query account akash1nxq8gmsw2vlz3m68qvyvcf3kh6q269ajvqw6y0
'@type': /cosmos.auth.v1beta1.BaseAccount
account_number: "13930"
address: akash1nxq8gmsw2vlz3m68qvyvcf3kh6q269ajvqw6y0
pub_key:
  '@type': /cosmos.crypto.secp256k1.PubKey
  key: Ay++nAhsGL7btFnlGxtiV8GlNQVjg9St9u66aVeMG7AA
sequence: "20474"

root@node1:~# kubectl -n akash-services logs $(kubectl -n akash-services get pods -l app=akash-provider --output jsonpath='{.items[0].metadata.name}') | grep 'syncing sequence' | tail -2
I[2022-06-12|09:31:15.243] syncing sequence                             cmp=client/broadcaster local=20474 remote=20474
I[2022-06-12|09:31:25.244] syncing sequence                             cmp=client/broadcaster local=20474 remote=20474
```

It would normally increment with every transaction, like in the following example `2738` incremented to `2739` after the TX:

```
akash-tools/cli-booster[https://rpc.akash.forbole.com:443][default][6284646-1-1]$ akash query account akash1h24fljt7p0nh82cq0za0uhsct3sfwsfu9w3c9h -o text
'@type': /cosmos.auth.v1beta1.BaseAccount
account_number: "14848"
address: akash1h24fljt7p0nh82cq0za0uhsct3sfwsfu9w3c9h
pub_key:
  '@type': /cosmos.crypto.secp256k1.PubKey
  key: Aj7abpFEMh8qIQ6PGgh6oD0lhovKAfgTfzEOiZTrV64r
sequence: "2738"

akash-tools/cli-booster[https://rpc.akash.forbole.com:443][default][6284646-1-1]$ akash_close 
INFO: Broadcasting 'akash deployment close -y' transaction...
INFO: Waiting for the TX A6EBC93FF8C70784F272B868FBDFBB48EA90CE5DE3A09661515584A4649FB3D1 to get processed by the Akash network
INFO: Success
6284646 deployment has been successfully closed.

akash-tools/cli-booster[https://rpc.akash.forbole.com:443][default][]$ akash query account akash1h24fljt7p0nh82cq0za0uhsct3sfwsfu9w3c9h -o text
'@type': /cosmos.auth.v1beta1.BaseAccount
account_number: "14848"
address: akash1h24fljt7p0nh82cq0za0uhsct3sfwsfu9w3c9h
pub_key:
  '@type': /cosmos.crypto.secp256k1.PubKey
  key: Aj7abpFEMh8qIQ6PGgh6oD0lhovKAfgTfzEOiZTrV64r
sequence: "2739"
```

**Comment by andy108369** [comment]
It looks like this has something to do with the account in relation to the RPC, as the account sequence started to increment as soon as I've broadcasted a TX with my provider's account to another RPC server:

```
root@akash-provider-545b8cf57b-pw6bp:/# echo $AKASH_NODE
http://akash-node-1:26657

root@akash-provider-545b8cf57b-pw6bp:/# akash query account akash1nxq8gmsw2vlz3m68qvyvcf3kh6q269ajvqw6y0
'@type': /cosmos.auth.v1beta1.BaseAccount
account_number: "13930"
address: akash1nxq8gmsw2vlz3m68qvyvcf3kh6q269ajvqw6y0
pub_key:
  '@type': /cosmos.crypto.secp256k1.PubKey
  key: Ay++nAhsGL7btFnlGxtiV8GlNQVjg9St9u66aVeMG7AA
sequence: "20474"

root@akash-provider-545b8cf57b-pw6bp:/# akash --node https://rpc.akash.forbole.com:443 tx deployment create deploy.yaml
{"height":"6284770","txhash":"216BF5CC1682DC99421EC85F94614FF732AF36F62B973F8AA2D4DF65B06EB8E7","codespace":"","code":0,"data":"0A2F0A2D2F616B6173682E6465706C6F796D656E742E763162657461322E4D73674372656174654465706C6F796D656E74",...

root@akash-provider-545b8cf57b-pw6bp:/# akash query account akash1nxq8gmsw2vlz3m68qvyvcf3kh6q269ajvqw6y0
'@type': /cosmos.auth.v1beta1.BaseAccount
account_number: "13930"
address: akash1nxq8gmsw2vlz3m68qvyvcf3kh6q269ajvqw6y0
pub_key:
  '@type': /cosmos.crypto.secp256k1.PubKey
  key: Ay++nAhsGL7btFnlGxtiV8GlNQVjg9St9u66aVeMG7AA
sequence: "20475"

root@akash-provider-545b8cf57b-pw6bp:/# akash tx deployment close --dseq 6284769 
{"height":"6284790","txhash":"EE90AB027AF2AE2931D8275B5AE537DA28272C89C8A27FC7D52BC97D4C7D30D3","codespace":"","code":0,"data":"0A2E0A2C2F616B6173682E6465706C6F796D656E742E763162657461322E4D7367436C6F73654465706C6F796D656E74",...
```

- original RPC is now incrementing the account sequence
```
root@akash-provider-545b8cf57b-pw6bp:/# akash query account akash1nxq8gmsw2vlz3m68qvyvcf3kh6q269ajvqw6y0
'@type': /cosmos.auth.v1beta1.BaseAccount
account_number: "13930"
address: akash1nxq8gmsw2vlz3m68qvyvcf3kh6q269ajvqw6y0
pub_key:
  '@type': /cosmos.crypto.secp256k1.PubKey
  key: Ay++nAhsGL7btFnlGxtiV8GlNQVjg9St9u66aVeMG7AA
sequence: "20476"

root@akash-provider-545b8cf57b-pw6bp:/# akash tx deployment create deploy.yaml
{"height":"6284801","txhash":"9221D54C93D03535E32B07C8BABB1E123B411CC4FBA8F0BAB1870A8437BD42E8","codespace":"","code":0,"data":"0A2F0A2D2F616B6173682E6465706C6F796D656E742E763162657461322E4D73674372656174654465706C6F796D656E74",....,{\"key\":\"dseq\",\"value\":\"6284800\"},....

root@akash-provider-545b8cf57b-pw6bp:/# akash query account akash1nxq8gmsw2vlz3m68qvyvcf3kh6q269ajvqw6y0
'@type': /cosmos.auth.v1beta1.BaseAccount
account_number: "13930"
address: akash1nxq8gmsw2vlz3m68qvyvcf3kh6q269ajvqw6y0
pub_key:
  '@type': /cosmos.crypto.secp256k1.PubKey
  key: Ay++nAhsGL7btFnlGxtiV8GlNQVjg9St9u66aVeMG7AA
sequence: "20477"

root@akash-provider-545b8cf57b-pw6bp:/# akash tx deployment close --dseq 6284800
{"height":"6284820","txhash":"7C4C3BA9B3134792F48286F00D8F3DAA0E39DB535F4AE64B6349460145E1A70A","codespace":"","code":0,....

root@akash-provider-545b8cf57b-pw6bp:/# akash query account akash1nxq8gmsw2vlz3m68qvyvcf3kh6q269ajvqw6y0
'@type': /cosmos.auth.v1beta1.BaseAccount
account_number: "13930"
address: akash1nxq8gmsw2vlz3m68qvyvcf3kh6q269ajvqw6y0
pub_key:
  '@type': /cosmos.crypto.secp256k1.PubKey
  key: Ay++nAhsGL7btFnlGxtiV8GlNQVjg9St9u66aVeMG7AA
sequence: "20478"
```

- provider logs
```
root@node1:~# kubectl -n akash-services logs $(kubectl -n akash-services get pods -l app=akash-provider --output jsonpath='{.items[0].metadata.name}') | grep 'syncing sequence' | tail -5
I[2022-06-12|09:41:35.244] syncing sequence                             cmp=client/broadcaster local=20475 remote=20475
I[2022-06-12|09:41:45.244] syncing sequence                             cmp=client/broadcaster local=20475 remote=20475
I[2022-06-12|09:41:55.244] syncing sequence                             cmp=client/broadcaster local=20475 remote=20475
I[2022-06-12|09:42:05.243] syncing sequence                             cmp=client/broadcaster local=20475 remote=20476
I[2022-06-12|09:42:15.243] syncing sequence                             cmp=client/broadcaster local=20476 remote=20476
```

**Comment by arno01** [comment]
// internal: potentially first appeared in 0.16.4-rc0; SUPPORT-105

**Comment by andy108369** [comment]
Haven't seen this issue since I've manually removed dead peers (`/root/bootstrap.sh` script) from our RPC nodes across Equinix (EWR1, SJC1, AMS1) last Friday on June 17th.

Have curated the peers, Equinix RPC nodes used, here https://github.com/ovrclk/net/pull/471

Lumen didn't encounter this issue as often as Equinix boxes since it did use a different source for peers https://github.com/ovrclk/helm-charts/blob/provider-0.177.0/charts/akash-node/values.yaml#L24

whereas only 1 peer is dead as of today (23 June 2022):

```
$ echo "$AKASH_P2P_PERSISTENT_PEERS" | tr ',' '\n' | tr ':' '@' | while IFS=@ read id ip port; do timeout 2s nc -vz $ip $port >/dev/null 2>&1 || echo ${id}@${ip}:${port} dead; done
15d1d26388ff693813a7c3c3ddc2b2e7382eb72e@202.61.197.65:26656 dead
```

The following command would show nothing for Equinix (EWR1, SJC1, AMS1) boxes:

```
journalctl -u akash-provider --no-pager -l --since '6 days ago' | grep -E 'account sequence mismatch|balance below target amount'
```

As well as it could be that something got fixed in `0.16.4` as we've upgraded from `0.16.4-rc2` to `0.16.4` on the same day - June 17th.

**Comment by andy108369** [comment]
No issue on our providers, closing.
